sql-parser/src/main/antlr/SqlBase.g4

```diff
@@ -527,6 +528,10 @@ setTypeDefinition
     : SET '(' dataType ')'
     ;
 
+tableTypeDefinition
+    : TABLE '(' dataType ')'
```

Comment:
How does the SQL look like where we need this to parse it?

Reply:
Would look like this:

```sql
Select * from where col = ANY(cast((Select col2 from t) as table(col_type)))
```

Whenever a cast is necessary, i.e. we're not selecting literals which can be converted without a cast.

Reply:
I'd like to avoid introducing a user-facing `table` type. Can we handle this in a different way?

Reply:
I'm afraid, I don't see how it can be avoided since TABLE is the real type which the subquery returns. If we convert from int->long or the other way, we need to apply a table cast.

Perhaps just get rid of the table type and use array? There is no difference currently between array and table, the table is represented as an array.